File: src/com/turlington/FileLoader.java
package com.turlington;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;
import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;

/**
 * Saves and loads jobs from text file.
 * Created by Mitchell on 7/18/2016.
 */
public class FileLoader {
    private final Path filePath = Paths.get("jobs.txt");

    Set<String> getJobListingsFromFile() {
        try {
            if (Files.exists(filePath)) {
                return Files.lines(filePath).collect(Collectors.toSet());
            }
            return new HashSet<>();
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.exit(1);
        return null;
    }

    void saveFile(JobListing jobListing) {
        try {
            Files.write(filePath, (jobListing.getTitle()+"\n").getBytes(), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
        } catch (IOException e) {
            e.printStackTrace(); //TODO: better error message
        }
    }
}

File: src/main/kotlin/com/turlington/FileLoader.kt
package com.turlington

import java.nio.file.Files
import java.nio.file.Paths
import java.nio.file.StandardOpenOption.APPEND
import java.nio.file.StandardOpenOption.CREATE
import java.util.stream.Collectors

/**
 * Saves and loads jobs from text file.
 * Created by Valerie on 7/18/2016.
 */
internal class FileLoader {
    private val filePath = Paths.get("jobs.txt")

    val jobListingsFromFile: Set<String>
        get() {
            if (Files.exists(filePath)) return Files.lines(filePath).collect(Collectors.toSet())
            return HashSet()
        }

    fun saveFile(jobListing: JobListing) {
        Files.write(filePath, ("${jobListing.title}\n").toByteArray(), CREATE, APPEND)
    }
}
